tools/capture_radar.rb

```diff
@@ -0,0 +1,49 @@
+Metric
+class Metric
+  def derived_cpu_total_cores_used
+    return if cpu_usage_rate_average.nil? || derived_vm_numvcpus.nil? || derived_vm_numvcpus == 0
+    (cpu_usage_rate_average * derived_vm_numvcpus) / 100.0
+  end
+end
+
+CustomAttribute
+class CustomAttribute
+  has_many   :metric_rollups, :as => :resource
+end
+
+labels = CustomAttribute.where(:name => "com.redhat.component")
+# Question - If a container is short lived, will we still be able to find it here?
+labels_containers = labels.each_with_object(Hash.new { |h,k| h[k] = [] }) { |l, h| h[l] += l.resource.containers.to_a }
+
+labels_metrics = Hash.new { |h,k| h[k] = [] }
+labels_containers.each do |l, containers|
+  containers.each do |c| 
+    labels_metrics[[l, c.container_project]] += c.metrics.to_a # This should be scoped to time range for the hour
```

Comment:
```labels_metrics[[l, c.container_project]].concat(c.metrics.to_a)```

otherwise it will always create a new array and it will take a lot of time for bigger amount of elements

Reply:
```c.metrics.where(:timestamp => (start..end)).maximum('cpu_usage_rate_average')```

And do the derived_cpu_total_cores_used calculation later. I could put together a windowed query that does the calculation for all metrics of all containers in 1 query.

Reply:
That would be awesome!! Thanks for the feedback @Ladas

Reply:
This tool doesn't have to be crazy performant...if it takes seconds to run that's ok...I'd rather we didn't put an inordinate amount of effort into it.